source/fuzz/transformation_add_type_struct.cpp

```diff
@@ -39,9 +39,11 @@ bool TransformationAddTypeStruct::IsApplicable(
   }
   for (auto member_type : message_.member_type_id()) {
     auto type = ir_context->get_type_mgr()->GetType(member_type);
-    if (!type || type->AsFunction()) {
-      // The member type id either does not refer to a type, or refers to a
-      // function type; both are illegal.
+    if (!type || type->AsFunction() ||
+        fuzzerutil::HasBlockOrBufferBlockDecoration(ir_context, member_type)) {
```

Comment:
Probably worth mentioning this in the docs.

Reply:
Similarly - in my working copy the comment is updated.

Reply:
(Same realisation as above.)